protos/protos/public/modeldb/Lineage.proto

```diff
@@ -39,18 +35,27 @@ message LineageEntryBatch {
 
 // Add every item of "input" as a parent of every item on "output".
 message AddLineage {
-    repeated LineageEntry input = 1;
-    repeated LineageEntry output = 2;
+    // If not provided, a new ID will be created
+    uint64 id = 1;
+
+    // An output can be associated with at most one lineage entry. That is the same as saying
+    // something can't be created by two separated paths.
+    // We also associated all inputs with all outputs. If the user wants more fine-grained control,
+    // they can create multiple entries.
+    repeated LineageEntry input = 2;
+    repeated LineageEntry output = 3;
+
 
     message Response {
-        bool status = 1;
+        uint64 id = 1;
     }
 }
 
 // Delete every item of "input" as a parent of every item on "output".
 message DeleteLineage {
-    repeated LineageEntry input = 1;
-    repeated LineageEntry output = 2;
+    uint64 id = 1; // Required
```

Comment:
is the design that we can partially delete a Lineage entry? if not then just the id should be sufficient, right?

Reply:
Yes, it should cover partial delete.